lib/grpc/GrpcServer.ts

```diff
@@ -44,10 +47,22 @@ class GrpcServer {
    * Start the server and begin listening on the provided port
    * @returns true if the server started listening successfully, false otherwise
    */
-  public listen = (port: number, host: string) => {
+  public listen = async (port: number, host: string, tlsCertPath: string, tlsKeyPath: string): Promise<boolean> => {
     assert(Number.isInteger(port) && port > 1023 && port < 65536, 'port must be an integer between 1024 and 65535');
 
-    const bindCode = this.server.bind(`${host}:${port}`, grpc.ServerCredentials.createInsecure());
+    if (!fs.existsSync(tlsCertPath)) {
+      this.logger.debug('Could not find gRPC TLS certificate. Generating new one');
+      this.generateCertificate(tlsCertPath, tlsKeyPath);
```

Comment:
`generateCertificate` writes files to disk, and then we read the files from disk later in this method. It could maybe return a  `{ cert_chain, private_key }` object to then be used below when we just generated the cert, otherwise read from disk.